lib/trento/domain/cluster/cluster.ex

```diff
@@ -469,6 +505,18 @@ defmodule Trento.Domain.Cluster do
     }
   end
 
+  defp maybe_emit_cluster_deregistered_event(
+         %Cluster{cluster_id: cluster_id, hosts_number: hosts_number},
```

Comment:
I don't think we can use the `hosts_number` here. This value is coming directly from the agent (parsed from crmmon or cibadmin) and it is not something we make up.
Here we need to get the list of `hosts` and get its length to do the comparison, which is the list we use to register the hosts in our aggregate.
We would need to check if we only have one host, and in that case trigger the deregistered event

Reply:
yes those are CIB details you can skip it